### PR TITLE
Mission Report

### PR DESCRIPTION
**Pull request by jkirk5**
### Summary
Addition of mission summary report as well as minor reworks to subsystem reports
- Adds a mission reporting framework that includes a simplified table of mission parameters, broken up by phase.
- Reworks propulsion report to query each engine to add to the report, as well as provide system-level metrics.
- Cleans up formatting for some existing subsystem reports
- Fixes some bugs related to Aviary reports not always being generated.

### Related Issues

None

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: aviary/interface/methods_for_level2.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 import warnings
 import inspect
 from pathlib import Path
+from copy import deepcopy
```
Comment (crecine) [inline aviary/interface/methods_for_level2.py:5]: unused?

File: aviary/interface/reports.py [hunk 2]
```diff
@@ -42,3 +52,118 @@ def subsystem_report(prob, **kwargs):
...
+                        vals = prob.get_val(f"{traj}.{phase}.{var_name}",
+                                            units=units,
+                                            indices=indices,)
+                    # 2DOF breguet range cruise uses time integration to track mass
+                    except TypeError:
+                        vals = prob.get_val(f"{traj}.{phase}.timeseries.time",
+                                            units=units,
+                                            indices=indices,)
+                    except KeyError:
+                        vals = None
```
Comment (johnjasa) [inline aviary/interface/reports.py:93]: Do all of these try/except blocks strictly need to be here? I'm thinking that most of the values we care about from the mission should be added to the timeseries (so they'd be accessible in the first try block).

Comment (crecine) [inline aviary/interface/reports.py:93]: could this be changed to something like:
```
key_paths = [f"timeseries.{var_name}", f"states:{var_name}", f"timeseries.states:{var_name}", f"{var_name}", "timeseries.time"]
vals = None
for key_path in key_paths:
	try:
		vals = prob.get_val(f"{traj}.{phase}.{key_path}", units=units, indices=indices,)
		return vals
	except KeyError:
		pass
```

File: aviary/interface/reports.py [hunk 3]
```diff
@@ -42,3 +52,118 @@ def subsystem_report(prob, **kwargs):
...
+    for idx, phase in enumerate(prob.phase_info):
+        # TODO for traj in trajectories, currently assuming single one named "traj"
+        # TODO delta mass and fuel consumption need to be tracked separately
+        fuel_burn = _get_phase_diff('traj', phase, 'mass', 'lbm', [-1, 0])
+        time = _get_phase_diff('traj', phase, 't', 'min')
+        range = _get_phase_diff('traj', phase, 'range', 'nmi')
+        # both "distance" and "range" currently exist to track ground distance, logic
+        # here accounts for that
+        if range is None:
+            range = _get_phase_diff('traj', phase, 'distance', 'nmi')
```
Comment (johnjasa) [inline aviary/interface/reports.py:122]: Do both distance and range still exist? I thought this PR resolved that by having just distance: https://github.com/OpenMDAO/Aviary/pull/73

**Review by johnjasa — CHANGES_REQUESTED** [review]
Looks slick! Thank you. I added some minor questions.

File: aviary/subsystems/propulsion/propulsion_builder.py [hunk 4]
```diff
@@ -61,14 +61,12 @@ def report(self, prob, reports_folder, **kwargs):
...
             write_markdown_variable_table(f, prob, propulsion_outputs, self.meta_data)
-            f.write('\n## ENGINES')
-            for engine in prob.aviary_inputs.get_val('engine_models'):
-                f.write(f'\n### {engine.name}')
-                write_markdown_variable_table(f, engine, engine_outputs, self.meta_data)
+            f.write('\n## Engines')
+
+        # each engine can append to this file
+        kwargs['meta_data'] = self.meta_data
+        for engine in prob.aviary_inputs.get_val('engine_models'):
```
Comment (crecine) [inline aviary/subsystems/propulsion/propulsion_builder.py:71]: This won't put in duplicates, right?
(a conventional twin engine aircraft would only have one entry)

**Review by crecine — APPROVED** [review]
One suggestion that may or may not be possible and a couple of minor comments

File: aviary/subsystems/propulsion/propulsion_builder.py (hunk 4 above)
Comment (jkirk5) [reply aviary/subsystems/propulsion/propulsion_builder.py:71]: Yeah, this is looping through each unique _type_ of engine

File: aviary/interface/reports.py (hunk 2 above)
Comment (jkirk5) [reply aviary/interface/reports.py:93]: That will work for everything except catching the TypeError that only shows up for the timeseries.time check. I'll play around a little with this idea

Comment (jkirk5) [reply aviary/interface/reports.py]: Good point. I did some tests and some of these aren't required. I've removed the unneeded try/except blocks

**Review by johnjasa — APPROVED** [review]

**Pull request merged**